draft-irtf-cfrg-hpke.md

```diff
@@ -428,37 +482,36 @@ def KeySchedule(mode, pkR, zz, enc, info, psk, pskID, pkSm):
   VerifyMode(mode, psk, pskID, pkSm)
 
   pkRm = Marshal(pkR)
-  identifier = "RFCXXXX"
   ciphersuite = concat(encode_big_endian(kem_id, 2),
                        encode_big_endian(kdf_id, 2),
                        encode_big_endian(aead_id, 2))
-  pskID_hash = Hash(pskID)
-  info_hash = Hash(info)
-  context = concat(identifier, ciphersuite, mode, enc, pkRm,
+  pskID_hash = LabeledExtract(0, "pskID", pskID)
+  info_hash = LabeledExtract(0, "info", info)
+  context = concat(ciphersuite, mode, enc, pkRm,
                    pkSm, pskID_hash, info_hash)
 
-  secret = Extract(psk, zz)
-  key = Expand(secret, concat("key", context), Nk)
-  nonce = Expand(secret, concat("nonce", context), Nn)
-  exporter_secret = Expand(secret, concat("exp", context), Nh)
+  if len(psk) > Nb then:
```

Comment:
Is it ok to use the already defined `len()` for the psk?
```
- `len(x)`: The length of the byte string `x`, expressed as a
  two-byte unsigned integer in network (big-endian) byte order
```
For now `len()` is only used to get the length of the nonce:
```
def Context.Nonce(seq):
  encSeq = encode_big_endian(seq, len(self.nonce))
  return xor(self.nonce, encSeq)
```
By the way, why compute the length of the nonce here? We know that it has length `Nn`. Also, why does encode_big_endian apparently need the length in “network (big-endian) byte order” and as two-byte integer?

Reply:
Yep, it's fine, and preferred! We pass a length in case the resulting number needs to be padded with zeros.